hathor/manager.py

```diff
@@ -344,11 +344,15 @@ def generate_mining_block(self, timestamp: Optional[float] = None,
         """
         from hathor.transaction.scripts import create_output_script
 
-        if not timestamp:
+        # We keep the original timestamp to use it in the `get_best_block_tips()` method,
+        # because this method has cache when `timestamp` is None.
+        original_timestamp = timestamp
+
+        if timestamp is None:
             timestamp = max(self.tx_storage.latest_timestamp, self.reactor.seconds())
```

Comment:
Isn't it easier to do this after we get the tip_blocks? This way there's no need to save `original_timestamp ` and the code is more readable.

Reply:
Great one! Fixed. Thanks!